### ISSUE TITLE
Selection of objects in 3d view does not work

### ISSUE THREAD

**Issue by mkeilman** [radiasoft/sirepo#3031]
For imported dump files, selection can only get the outermost group, not its constituents. For user-built magnets, selection fails completely.  There is a breakdown in the mapping of VTK "actors" to Radia objects

**Comment by mkeilman** [comment]
This has been ameliorated somewhat since the issue was created in that selection works properly for individual objects. However, we still need to get at objects inside groups.

**Comment by mkeilman** [comment]
@gurhar1133 this is an interesting one. Selection of 3d objects in radia happens here:

https://github.com/radiasoft/sirepo/blob/de5f93a18f57b610d87363096b351d825de5fa6e/sirepo/package_data/static/js/radia.js#L2975
...
https://github.com/radiasoft/sirepo/blob/de5f93a18f57b610d87363096b351d825de5fa6e/sirepo/package_data/static/js/radia.js#L3087

To expand on the issue, the objects are almost always in a group (or "container" in Radia terms), with the result that when a user selects an object it highlights the entire group. The goal of this issue is to drill down to get the closest object within the group, which is complicated by the fact that groups are often nested. Work with the Undulator example, which has a number of groups to deal with.

**Comment by gurhar1133** [comment]
Do you have a example file/sim I could replicate issue with @mkeilman ?

**Comment by mkeilman** [comment]
> Do you have a example file/sim I could replicate issue with @mkeilman ?

Yes, "Undulator" under Examples is a good one. Click the arrow button under the viewer and control-click say a light blue object. You'll see 

<img width="704" alt="Screen Shot 2022-07-12 at 12 13 57 PM" src="https://user-images.githubusercontent.com/30352290/178565379-812e6d4e-f819-4cd7-936e-b5acec82db78.png">

"octant" is the top-level group. Note that if you now pick a color, every object gets that color. What we want is to get a permanent magnet as defined under Design/Permanent Magnets.

**Comment by mkeilman** [comment]
Quick ping @gurhar1133 to see if anything needs clarification. No rush.

**Comment by gurhar1133** [comment]
> Quick ping @gurhar1133 to see if anything needs clarification. No rush.

Thanks for checking in @mkeilman :) Been working on nuline stuff lately since I am low on mpmd hours, but I will reach out when I revisit.

**Comment by mkeilman** [comment]
@gurhar1133  I think @cchall said you were getting 40 mpmd hours this month (not final, he can correct me) so wanted to ping. I'm working on a wiki for the code as well.

**Comment by gurhar1133** [comment]
@mkeilman I'll bump this up in priority then
